# Assignment 6

To measure the psychological state of Americans, a psychology researcher (in 1929) devised
three questions put to 3443 Americans from different walks of life. He asked them whether they
were happy. They could answer only **yes** or **no** to each question. By today’s standards the study
is quite simple, and he did no formal statistical analysis of his data. The frequency data (number
of **yes** and **no** responses within each employment category) from this study are in the file
*Americandata.csv*.

---

You need to use the data and answer the question, that is, **Is the type of profession related to the happiness of American people?**
Report the **χ2** statistic and the *statistical significance of test*, in addition to the contingency table.
Also, explain the results and what they mean. What does the value of **χ2** statistic mean? Is it
good or bad? Is the test statistically significant?

In [25]:
import pandas as pd
import scipy.stats as stats

df = pd.read_csv("./Americandata.csv", encoding="latin-1")

df_fixed = df.melt(id_vars=["Happy"], var_name="Profession", value_name="Count")

contingency_table = df_fixed.pivot(index="Profession", columns="Happy", values="Count") 
print("====Contingency table====")
print(contingency_table)

chi2, p_value, dof, expected = stats.chi2_contingency(contingency_table, correction=False) 
print("\n====Chi square thing====")
print(f"Chi-square = {chi2:.2f}")
print(f"Degrees of freedom = {dof}")
print(f"p-value = {p_value:.2e}")

====Contingency table====
Happy         No  Yes
Profession           
Housewives   122   78
Laborers     122  378
Lawyers       64   11
Musicians     19   31
Physicians    51  159
Preachers    265   35
Students    1610  390
Teachers      38  108

====Chi square thing====
Chi-square = 936.14
Degrees of freedom = 7
p-value = 7.52e-198


Put your analysis in the empty `markdown` cell below.

The results show the most unhappy profession are either students or preachers. The Chi-Square value measures how much a data set differs from what you would expect. A value of 936.14 is not good, and it means it differs from the mean by a whole lot. This also means it is reliant on the degrees of freedom. There is a huge mismatch and with a low p-value this means we reject the null hypothesis